includes/functions.php

```diff
@@ -173,6 +173,327 @@ function url_to_authorid( $url ) {
+	return $in_reply_to;
+}
+
+/**
+ * Checks a comment ID for a source_id, or source_url
+ */
+function get_source_id( $comment ) {
+
+	if ( $comment->user_id !== 0 ) {
+
+		$source_id = \get_comment_meta( $comment->ID, 'source_id', true );
+		if ( ! $source_id ) {
+			$source_url = \get_comment_meta( $comment->ID, 'source_url', true );
+			if ( ! $source_url ) {
+				return null;
+			}
+			$response = safe_remote_get( $source_url );
+			$body = \wp_remote_retrieve_body( $response );
+			$remote_status = \json_decode( $body, true );
+			if ( is_wp_error( $remote_status )
+				|| ! isset( $remote_status['@context'] )
+				|| ! isset( $remote_status['object']['id'] ) ) {
+
+				// the original post may have been deleted, before we started processing deletes.
+				return null;
+			}
+			$source_id = $remote_status['object']['id'];
+		}
+		return $source_id;
+	}
+	return null;
+}
+
+/**
+ * Verify if url is a wp_ap_comment,
+ * Or if it is a previously received remote comment
+ * @return int comment_id
+ */
+function is_ap_comment() {
+	$comment_id = get_query_var( 'replytocom', null );
+	if ( ! is_null( $comment_id ) ) {
+		$comment = \get_comment( $comment_id );
+		// Only return local origin comments
+		if ( $comment->user_id ) {
+			return $comment_id;
+		}
+	}
+	return null;
+}
+
+/**
+ * Verify if url has a replies query,
+ * @return bool
+ */
+function is_ap_replies() {
+	$replies = get_query_var( 'replies' );
+	if ( $replies ) {
+		return $replies;
+	}
+	return null;
+}
+
+/**
+ * Get recipients to insert/tag in reply tag from received AP object meta
+ * @param string $object_id a comment_id to search
+ * @param boolean $post defaults to searching a comment_id
+ * @return string space separated webfinger of tagged users
+ */
+function reply_recipients( $object_id, $post = null ) {
+	$recipients = null;
+	if ( $post ) {
+		//post
+		$ap_object = \unserialize( \get_post_meta( $object_id, 'ap_object', true ) );
+	} else {
+		//comment
+		$ap_object = \unserialize( \get_comment_meta( $object_id, 'ap_object', true ) );
+	}
+
+	if ( ! empty( $ap_object ) ) {
+		// Replying to remote comments.
+		$recipients[] = \ActivityPub\url_to_webfinger( $ap_object['actor'] ); // Reply to Object actor!
+
+		if ( ! empty( $ap_object['object']['tag'] ) ) { // Reply to other tagged users.
+			$author_post_url = \get_author_posts_url( $ap_object['user_id'] );// ignore self tag.
+			foreach ( $ap_object['object']['tag'] as $tag ) { // Other tagged users
+				if ( $author_post_url === $tag['href'] ) {
+					continue;
+				}
+				if ( in_array( 'Mention', $tag ) ) {
+					$recipients[] = $tag['name'];
+				}
+			}
+		}
+		return implode( ' ', $recipients );
+	} else {
+		// Replying to self with others.
+		$comment = \get_comment( $object_id );
+		preg_match_all( '/@' . ACTIVITYPUB_USERNAME_REGEXP . '/', $comment->comment_content, $recipients );
+		return implode( ' ', $recipients[0] );
+	}
+}
+
+/**
+ * Add summary to reply
+ */
+function get_summary( $comment_id ) {
```

Comment:
this is a very generic name for a Comment only thing, maybe also move to Comment Class?